modules/mono/csharp_script.cpp

```diff
@@ -977,12 +977,24 @@ void CSharpLanguage::reload_assemblies(bool p_soft_reload) {
 #endif
 
 		if (!scr->get_path().is_empty()) {
-			scr->reload(p_soft_reload);
+			// Constructing default properties while reloading the assembly can lead to their scripts being created and registered by this point
+			// Check if a script was created that way and let it take over the path rather than trying to reload this script which would fail
+			Ref<CSharpScript> existing_script;
+			String script_path = scr->get_path();
+
+			if (GDMonoCache::godot_api_cache_updated && GDMonoCache::managed_callbacks.ScriptManagerBridge_GetExistingScriptBridgeForPath(&script_path, &existing_script)) {
+				existing_script->pending_reload_instances = scr->pending_reload_instances;
+				existing_script->pending_reload_state = scr->pending_reload_state;
+				existing_script->_take_over_path(script_path);
```

Comment:
Is this necessary here? I think it'd be an issue if we end up with `CSharpScript`s in the bimap, but don't have the right path on the resource already?

Reply:
I believe the important part here is that we're taking over the path from the other script, since we otherwise have both of them pointing to the same path which I think was leading to asserts.